src/server/pfs/server/driver.go

```diff
@@ -558,35 +568,33 @@ func (d *driver) makeCommit(pachClient *client.APIClient, ID string, parent *pfs
 					return err
 				}
 			}
-		} else {
-			isStartCommit = true
-			if err := d.openCommits.ReadWrite(stm).Put(newCommit.ID, newCommit); err != nil {
-				return err
-			}
-		}
 
-		if treeRef != nil {
+			// now 'treeRef' is guaranteed to be set
 			newCommitInfo.Tree = treeRef
 			newCommitInfo.SizeBytes = uint64(tree.FSSize())
 			newCommitInfo.Finished = now()
-		}
 
-		// Update the repo size if we're on the master branch and this is not
-		// an open commit
-		if !isStartCommit && branch == "master" {
-			repoInfo.SizeBytes = newCommitInfo.SizeBytes
+			// If we're updating the master branch, also set the repo size (persisted
+			// below, along with new branch)
+			if branch == "master" {
+				repoInfo.SizeBytes = newCommitInfo.SizeBytes
+			}
```

Comment:
it looks like we're no longer checking if records is nil before updating the repoInfo. Is this correct?

Reply:
See below—I believe the code should still work this way